File: easy/0905_sort_array_by_parity/solution.py
from typing import List


class Solution:
    def sortArrayByParity(self, nums: List[int]) -> List[int]:
        even = 0

        for i in range(len(nums)):
            if nums[i] % 2 == 0:
                nums[even], nums[i] = nums[i], nums[even]
                even += 1

        return nums

File: easy/0905_sort_array_by_parity/solution.cpp
#include <vector>

class Solution {
public:
  std::vector<int> sortArrayByParity(std::vector<int> &nums) {
    int even = 0;

    for (size_t i = 0; i < nums.size(); ++i) {
      if (nums[i] % 2 == 0) {
        std::swap(nums[even], nums[i]);
        ++even;
      }
    }

    return nums;
  }
};
